test/iobuffer.jl

```diff
@@ -348,3 +348,12 @@ end
 @testset "bytesavailable devnull" begin
     @test bytesavailable(devnull) == 0
 end
+
+@testset "#48188 read_sub for non Array AbstractArray" begin
+    a = [0,0,0]
+    v = @view a[1:2]
+    io = IOBuffer()
+    write(io,1)
+    seek(io,0)
+    @test Base.read_sub(io,v,1,1) == [1,0]
```

Comment:
If there is no surface API that reaches this, perhaps the `else` block should just be removed?

Reply:
In fact since it's only used in a single place we might even want to inline it in readbytes!

Reply:
yeah, we switched to unsafe_read, so go for it

Reply:
It makes the code uglier so I guess it's fine. This is good to go IMO